Src/AmrCore/AMReX_FillPatcher.H

```diff
@@ -55,6 +55,17 @@ namespace amrex {
  * allowed to fill only some of the components with the fill function.
  *
  * (5) This only works for cell-centered and nodal data.
+ *
+ * This class also provides support for RungeKutta::RK3 and RungeKutta::RK4.
+ * The storeRKCoarseData function can be used to store coarse AMR level
+ * data that are needed for filling fine level data's ghost cells in this
+ * class.  The `fillRK` function can be used to fill ghost cells for fine
+ * AMR levels.  This operation at the coarse/fine boundary is non-trivial
+ * for RK orders higher than 2.  Note that it is expected that time stepping
```

Comment:
Can you add a comment here or in the fillRK doxygen about how to do that trivial RK2 ghost cell fill? Is that the functionality demonstrated in the CNS test modifications in this PR? Is there a reason to not have this class support RungeKutta::RK2 as well for completeness?

Reply:
The reason I don't want to make it support RK2 is that, for RK2, we don't need to call `storeRKCoarseData`.  So I think it might add more confusion if it supports RK2.  Also, filling ghost cells for RK2 can be done with the normal fillpath functions.  For the CNS code, the RK part of the code is actually very clean because it's based on AmrLevel.
```
    RK(rk_order, State_Type, time, dt, iteration, ncycle,
       // Given state S, compute dSdt. dtsub is needed for flux register operations                                                                                          
       [&] (int /*stage*/, MultiFab& dSdt, MultiFab const& S, Real dtsub) {
           compute_dSdt(S, dSdt, dtsub, fr_as_crse, fr_as_fine);
       },
       // Optional. In case if there is anything needed after each RK substep.                                                                                               
       [&] (int /*stage*/, MultiFab& S) { computeTemp(S,0); });
```
This works for RK2, 3 and 4.